src/test/scala/dev/kamu/engine/spark/ingest/EngineIngestTest.scala

```diff
@@ -252,4 +260,71 @@ class EngineIngestTest extends FunSuite with KamuDataFrameSuite with Matchers {
     )
   }
 
+  test("ingest Parquet") {
+    Temp.withRandomTempDir("kamu-engine-spark")(
+      tempDir => {
+        val outputLayout = tempLayout(tempDir, "out")
+
+        val inputPath = tempDir.resolve("input1.parquet")
+        ParquetHelpers.write(
+          inputPath,
+          Seq(
+            CityEvent(LocalDate.parse("2020-01-01"), "A", 1000),
+            CityEvent(LocalDate.parse("2020-02-01"), "B", 2000)
+          )
+        )
+
+        val outputPath = outputLayout.dataDir.resolve("pending.snappy.parquet")
+
+        val request = yaml.load[IngestRequest](
+          s"""
+             |datasetID: "did:odf:abcd"
+             |datasetName: out
+             |ingestPath: "${inputPath}"
+             |systemTime: "2020-01-01T00:00:00Z"
+             |eventTime: "2020-01-01T00:00:00Z"
+             |offset: 10
+             |source:
+             |  fetch:
+             |    kind: url
+             |    url: http://localhost
+             |  read:
+             |    kind: parquet
+             |  merge:
+             |    kind: append
+             |datasetVocab: {}
+             |prevCheckpointPath: null
+             |newCheckpointPath: "${outputLayout.checkpointsDir}"
+             |dataDir: "${outputLayout.dataDir}"
+             |outDataPath: "${outputPath}"
+             |""".stripMargin
+        )
+
+        val engineRunner = new EngineRunner(new DockerClient)
+        val response = engineRunner.ingest(request, tempDir)
+
+        response.dataInterval.get shouldEqual OffsetInterval(
+          start = 10,
+          end = 11
+        )
+        response.outputWatermark shouldEqual Some(
+          Instant.parse("2020-01-01T00:00:00Z")
```

Comment:
I was surprised to see that the expected watermark is `2020-01-01` instead of `2020-02-01` (your last event date).

I believe this is because your test will currently produce dataset with schema:
`offset | system_time | event_time | date | city | population`

> Note there will be both `date` and `event_time`.

That's because you don't specify this to use `date` as your event time:
```yaml
datasetVocab:
  eventTimeColumn: date
```
The `append` merge strategy thinks that this dataset doesn't have event time in it and adds it artificially using the `eventTime` fallback you provided in the request.

Better to make proper use of the `date` column, otherwise this test may look confusing.

Reply:
Corrected. Haven't noticed that header, although I was copying CSV tests as a base reference.